platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoCoverageVerification.java

```diff
@@ -92,8 +94,12 @@ public void check() throws IOException {
             parameters.getAllClassesDirs().convention(getAllClassDirs());
             parameters.getExecutionData().convention(getExecutionData());
 
-            parameters.getFailOnViolation().convention(getViolationRules().isFailOnViolation());
-            parameters.getRules().convention(getViolationRules().getRules());
+            parameters.getFailOnViolation().convention(getViolationRules().getFailOnViolation());
+            parameters.getRules().convention(
+                getViolationRules().getRules().get().stream()
+                    .map(SerializableJacocoViolationRule::of)
```

Comment:
💭 I have to map them to a new type without `Property` types, otherwise worker serialization fails, since `Property` types are not serializable